docs/parameters.md

```diff
@@ -0,0 +1,188 @@
+
+Let's say we want to create an `Instance` of EC2 running on AWS using a
+[corresponding broker](https://github.com/cloudfoundry-samples/go_service_broker) 
+which implements the Open Service Broker API.
+
+A typical provisioning request for this broker looks [like this](https://github.com/cloudfoundry-samples/go_service_broker/blob/master/bin/curl_broker.sh):
+```bash
+curl -X PUT http://username:password@localhost:8001/v2/service_instances/instance_guid-111 -d '{
+  "service_id":"service-guid-111",
+  "plan_id":"plan-guid",
+  "organization_guid": "org-guid",
+  "space_guid":"space-guid",
+  "parameters": {"ami_id":"ami-ecb68a84"}
+}' -H "Content-Type: application/json"
+```
+
+Note that the broker accepts an `ami_id` parameter ([AMI](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html) 
+identifier).
+To configure a provisioning request in Service Catalog, we need to declare an `Instance` 
+resource with an AMI identifier declared in the `parameters` field of its spec:
+```yaml
+apiVersion: servicecatalog.k8s.io/v1alpha1
+kind: Instance
+metadata:
+  name: ami-instance
+  namespace: test-ns
+spec:
+  serviceClassName: aws-ami
+  planName: default
+  parameters:
+    ami_id: ami-ecb68a84
+```
+
+### Passing sensitive data
+
+`Secret` resources can be used to store sensitive data. The `parametersFrom`
+field allows to reference the external parameters source.
+The following sections describe the two ways in which secrets can be used to 
+populate the parameters sent to a broker.
+
+#### Reference to a secret
+
+Use the `secretRef` field to pass a reference to a secret with parameter contents.
+
+```yaml
+apiVersion: servicecatalog.k8s.io/v1alpha1
+kind: Instance
+metadata:
+  name: qwerty-instance
+  namespace: test-ns
+spec:
+  serviceClassName: qwerty
+  planName: default
+  parametersFrom:
+    - secretRef:
+        name: mysecret
+```
+
+Each secret key will be transformed into a top-level parameter name, and 
+corresponding data will be represented as a value for this parameter.
+
+Note that every value is always treated as a **string**, other value types are 
+not supported for `secretRef`.
+
+#### The entire payload with sensitive data
+
+If the user has the entire JSON payload prepared to be sent, and it 
+contains sensitive data, this payload can be stored in a single `Secret` key, and 
+passed using a `secretKeyRef` field:
+
+```yaml
+  ...
+  parametersFrom:
+    - secretKeyRef:
+        name: mysecret
+        key: mykey
+```
+
+The value stored in a secret key must be a valid JSON.
+
+### Merging multiple sources and conflict resolution
+
+If multiple sources in `parameters` and `parametersFrom` blocks are specified,
+the final payload is a result of merging all of them at the top level.
+If there are any duplicate properties defined at the top level, the specification
+is considered to be invalid, the further processing of the `Instance`/`Binding` 
+resource stops and its `status` is marked with error condition.
```

Comment:
I think whether duplicate properties are an error depends on what the user's preference is.  We have precedent for overriding based on declaration precedence in other areas of the system.  That said, I'm willing to make the initial behavior fail-fast in order to unblock this - we can add differentiation of override vs. fail-fast behaviors in a backward-compatible way later.